src/Adapter/Customer/QueryHandler/GetCustomerInformationHandler.php

```diff
@@ -0,0 +1,590 @@
+     */
+    private function getCustomerRankBySales($customerId)
+    {
+        $sql = 'SELECT SUM(total_paid_real) FROM ' . _DB_PREFIX_ . 'orders WHERE id_customer = ' . (int) $customerId . ' AND valid = 1';
+
+        if ($totalPaid = Db::getInstance()->getValue($sql)) {
+            $sql = '
+                SELECT SQL_CALC_FOUND_ROWS COUNT(*)
+                FROM ' . _DB_PREFIX_ . 'orders
+                WHERE valid = 1
+                    AND id_customer != ' . (int) $customerId . '
+                GROUP BY id_customer
+                HAVING SUM(total_paid_real) > ' . (int) $totalPaid
+            ;
+
+            Db::getInstance()->getValue($sql);
+
+            return (int) Db::getInstance()->getValue('SELECT FOUND_ROWS()') + 1;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param Customer $customer
+     *
+     * @return OrdersInformation
+     */
+    private function getCustomerOrders(Customer $customer)
+    {
+        $validOrders = [];
+        $invalidOrders = [];
+
+        $orders = Order::getCustomerOrders($customer->id, true);
+        $totalSpent = 0;
+
+        foreach ($orders as $order) {
+            $order['total_paid_real_not_formated'] = $order['total_paid_real'];
+            $order['total_paid_real'] = Tools::displayPrice(
+                $order['total_paid_real'],
+                new Currency((int) $order['id_currency'])
+            );
+
+            if (!isset($order['order_state'])) {
+                $order['order_state'] = $this->translator->trans(
+                    'There is no status defined for this order.',
+                    [],
+                    'Admin.Orderscustomers.Notification'
+                );
+            }
+
+            $customerOrderInformation = new OrderInformation(
+                (int) $order['id_order'],
+                Tools::displayDate($order['date_add']),
+                $order['payment'],
+                $order['order_state'],
+                (int) $order['nb_products'],
+                $order['total_paid_real']
+            );
+
+            if ($order['valid']) {
+                $validOrders[] = $customerOrderInformation;
+                $totalSpent += $order['total_paid_real_not_formated'] / $order['conversion_rate'];
+            } else {
+                $invalidOrders[] = $customerOrderInformation;
+            }
+        }
+
+        return new OrdersInformation(
+            Tools::displayPrice($totalSpent),
+            $validOrders,
+            $invalidOrders
+        );
+    }
+
+    /**
+     * @param Customer $customer
+     *
+     * @return CartInformation[]
+     */
+    private function getCustomerCarts(Customer $customer)
+    {
+        $carts = Cart::getCustomerCarts($customer->id);
+        $customerCarts = [];
+
+        foreach ($carts as $cart) {
+            $cart = new Cart((int) $cart['id_cart']);
+            Context::getContext()->cart = $cart;
```

Comment:
What is the purpose of setting `Context::getContext()->cart` in this `foreach` loop ?

Reply:
to make legacy things work. :D i know it sucks, but in this case it must be set to make  `$cart->getSummaryDetails();` work. :/